### PR TITLE
Adding error handling to publish errors in table designer.

### LINKED ISSUE #18166: Table Designer: Error not reported during publishing, causing infinite loading "Publishing Changes"
Steps to repro:
1. Connect to a db and start table designer to add a new table
2. Shutdown the db before clicking "Update Database"
3. Click "Update Database"

No error is reported, and the spinner loads infinitely.
image
I was able to close the publishing dialog by pressing Esc multiple times, not sure if this intended design as well.

We can also see the same behavior for any script error during publishing (I was able to repro the same issue with memory-optimized table).

ADS reports back the error.
image

### PR DESCRIPTION
**Pull request by aasimkhan30**
image

This fixes #18166

### CODE REVIEW

**Comment by caohai** [comment]
Can we also add something like a `Copy Details` button that allow user to copy the error message?

**Comment by cssuh** [comment]
Ignore the tag from my PR, it was an accident

File: src/reactviews/common/locConstants.ts [hunk 1]
```diff
@@ -132,6 +133,8 @@ export class LocConstants {
                     args: [objectType],
                     comment: ["{0} is the object type"],
                 }),
+            retryUpdatingDatabase: l10n.t("Retry updating database"),
```
Comment (Benjin) [inline src/reactviews/common/locConstants.ts:136]: The text for these buttons are a little long.  Maybe something simpler like "Retry" and "Back to preview".

@jdesai-24

**Comment by Benjin** [comment]
That `( ! )` icon is huge.  What about displaying the error with a <MessageBar> like this dialog does?  #18147

File: src/reactviews/pages/TableDesigner/designerChangesPreviewButton.tsx [hunk 2]
```diff
@@ -130,13 +146,48 @@ export const DesignerChangesPreviewButton = () => {
                 </>
             );
         }
+        if (metadata?.apiState?.publishState === LoadState.Error) {
```
Comment (Benjin) [inline src/reactviews/pages/TableDesigner/designerChangesPreviewButton.tsx:149]: these LoadState cases feel like they're screaming for refactoring the dialog content out into helper functions/components.

File: src/reactviews/common/locConstants.ts (hunk 1 above)
Comment (jdesai-24) [reply src/reactviews/common/locConstants.ts:136]: @Benjin, I agree, we can add "Retry" and "Back to preview". Do we need a close button, I think a cross icon would suffice. 
For the error message, can we break down the message into different points which will be easier to read?

**Review by caohai — APPROVED** [review]

**Review by lewis-sanchez — APPROVED** [review]

**Pull request merged**